### PR TITLE
Add eslint-plugin-regexp

### PR DESCRIPTION
**Pull request by silverwind**
Add [`eslint-plugin-regexp`](https://github.com/ota-meshi/eslint-plugin-regexp) and fix discovered issues. Config is mostly the recommended one, but I relaxed a few rules.

### CODE REVIEW

**Comment by yardenshoham** [comment]
Do the changes you made have any consequences? (`gm` -> `g`, removing `i`)

**Comment by silverwind** [comment]
No. Errors were:

```
docs/assets/js/search.js
  141:78  error  Unescaped source character '}'  regexp/strict

web_src/js/features/codeeditor.js
  139:82  error  The 'i' flag is unnecessary because the pattern only contains case-invariant characters  regexp/no-useless-flag

web_src/js/utils.js
  35:34  error  The 'm' flag is unnecessary because the pattern does not contain start (^) or end ($) assertions  regexp/no-useless-flag
```

The two flag cases where autofixed actually, the unescaped `}` is the only potentially breaking one but I verified it matches what I think it matches before and after: https://regex101.com/r/AjKChp/1

**Review by yardenshoham — APPROVED** [review]

File: web_src/js/features/codeeditor.js [hunk 1]
```diff
@@ -136,7 +136,7 @@ function togglePreviewDisplay(previewable) {
   if (!previewTab) return;
 
   if (previewable) {
-    const newUrl = (previewTab.getAttribute('data-url') || '').replace(/(.*)\/.*/i, `$1/markup`);
```
Comment (delvh) [inline web_src/js/features/codeeditor.js:139]: Was the `i` removed as this regex does not care about case at all?

File: web_src/js/utils.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ export function isDarkTheme() {
 
 // strip <tags> from a string
 export function stripTags(text) {
-  return text.replace(/<[^>]*>?/gm, '');
+  return text.replace(/<[^>]*>?/g, '');
```
Comment (delvh) [inline web_src/js/utils.js:35]: Why no multiline anymore?

File: web_src/js/features/codeeditor.js (hunk 1 above)
Comment (yardenshoham) [reply web_src/js/features/codeeditor.js:139]: Look at https://github.com/go-gitea/gitea/pull/24361#issuecomment-1523865177

**Review by delvh — APPROVED** [review]

**Pull request merged**